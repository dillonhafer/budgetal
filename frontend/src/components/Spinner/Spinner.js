import React from 'react';
import PropTypes from 'prop-types';
import { Pane, Text, Spinner as GreenSpinner } from 'evergreen-ui';

const Spinner = ({ visible = false, title = 'Loading...' }) =>
  visible ? (
    <Pane textAlign="center" marginY={56}>
      <GreenSpinner marginX="auto" />
      <Text marginY={16}>{title}</Text>
    </Pane>
  ) : null;

Spinner.propTypes = {
  visible: PropTypes.bool,
  title: PropTypes.string,
};

export default Spinner;
