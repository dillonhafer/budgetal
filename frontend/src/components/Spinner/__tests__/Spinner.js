import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '../Spinner';

describe('Spinner', () => {
  describe('when visible is true', () => {
    it('renders a spinner', () => {
      const comp = shallow(<Spinner visible={true} />);
      expect(
        comp
          .find('withTheme(Text)')
          .children()
          .text(),
      ).toEqual('Loading...');
      expect(comp).toMatchSnapshot();
    });
  });

  describe('when visible is false', () => {
    it('renders null', () => {
      const comp = shallow(<Spinner visible={false} />);
      expect(comp.type()).toEqual(null);
    });
  });

  it('renders custom text', () => {
    const comp = shallow(<Spinner visible={true} title="Hold On..." />);
    expect(
      comp
        .find('withTheme(Text)')
        .children()
        .text(),
    ).toEqual('Hold On...');
  });
});
