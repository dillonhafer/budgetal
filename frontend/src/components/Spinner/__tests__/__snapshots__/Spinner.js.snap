// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner when visible is true renders a spinner 1`] = `
<withTheme(Pane)
  marginY={56}
  textAlign="center"
>
  <withTheme(Spinner)
    marginX="auto"
  />
  <withTheme(Text)
    marginY={16}
  >
    Loading...
  </withTheme(Text)>
</withTheme(Pane)>
`;
